code/modules/mob/living/carbon/human/death.dm

```diff
@@ -61,7 +61,11 @@
 	animation.master = src
 
 	flick("dust-h", animation)
-	new /obj/effect/decal/remains/human(loc)
+	new species.remains_type(get_turf(src))
+	if(isSynthetic())
+		var/datum/effect/system/spark_spread/s = new /datum/effect/system/spark_spread
+		s.set_up(3, 1, src)
+		s.start()

```

Comment:
Instead of an `isSynthetic()` check, you could just make a subtype of `/obj/effect/decal/remains/robot` that spawns sparks in its `New()` and then calls `..()`